uber/models/department.py

```diff
@@ -312,18 +341,35 @@ class Job(MagModel):
         secondary='job_required_role')
     shifts = relationship('Shift', backref='job')
 
-    _repr_attr_names = ['name']
+    __table_args__ = (
+        Index('ix_job_department_id', department_id),
+    )
 
-    # Using a column_property for department_name *might* be more efficient,
-    # due to our prevelant usage of the department_name property.
-    department_name = column_property(
-        select([Department.name], Department.id == department_id))
+    _repr_attr_names = ['name']
 
     @classproperty
     def extra_apply_attrs(cls):
         return set(['required_roles_ids']).union(
             cls.extra_apply_attrs_restricted)
 
+    @hybrid_property
+    def department_name(self):
+        return self.department.name
+
+    @department_name.expression
+    def department_name(cls):
+        return select([Department.name]).where(
+            Department.id == cls.department_id).label('department_name')
```

Comment:
I couldn't quite understand the SQLAlchemy docs on this -- what is `label` doing?

Reply:
The `.label()` is adding an `AS` statement to the generated SQL. The easiest way to explain it is probably to show you the SQL that is actually generated!

```
 SELECT job.id, department.name AS department_name FROM job, department WHERE job.department_id = department.id;
```

And the results of that query will look like this:
```
                  id                  |  department_name  
--------------------------------------+-------------------
 96f4b4f5-e3f9-41d1-8ab7-209887242e57 | Arcade
 e486fa81-008a-40c0-aa7d-bd1f99fd592d | Tech Ops
 469c6808-ef5d-4c5c-8ee2-65469cb16629 | Consoles
 99cdf401-008a-4f89-81df-19736b96e822 | Panels
 11deb14e-7dfb-4735-b048-edd2e7d56415 | Staff Suite
```

I don't know if that's strictly necessary anymore. But it's nice when the the returned column names map to the property names!